src/script/engine.zig

```diff
@@ -164,53 +185,17 @@ pub const Engine = struct {
         self.pc += n;
     }
 
-    /// OP_PUSHDATA1: Push the next byte as N, then push the next N bytes
-    ///
-    /// # Returns
-    /// - `EngineError`: If an error occurs during execution
-    fn opPushData1(self: *Engine) !void {
-        if (self.pc + 1 > self.script.len()) {
+    /// Generalized function to handle OP_PUSHDATA operations with variable length prefixes.
+    /// Pushes the next `n_bytes` as a length, then pushes that many bytes as an array.
+    fn opPushDataX(self: *Engine, n_bytes: usize) !void {
```

Comment:
There is no such an opcode.

The only valid n_bytes value are 1, 2 and 4. This implementation allows for any number of bytes to be pushed.

Please revert to the previous 3 distinct functions.

If you really want to factorize code, do it internally, but keep exposing 3 different methods. 1 for each opcode

Reply:
Alright, I'll.

Reply:
reverted